external-testing/README.md

```diff
@@ -0,0 +1,65 @@
+# External Testing
+
+Let's talk about testing process against external resources. What does that
+mean? Well, talking to GitHub API, COPR API, running `fedpkg`, cloning
+repositories...
+
+
+## Use cases
+
+* Interact with an external API (such as GitHub, COPR, Pagure, FAS, k8s)
+  * Authenticated vs. unauthenticated calls.
+* Clone a git repository.
+* Construct a git repository.
+* Run a command to perform an action.
+  * Changes happening only locally.
+  * RO access to a remote service.
+  * R/W access to a remote service.
+* Combination of multiple use cases into a single test case.
+
+
+## Solutions
+
+* flexmock
+
+* pytest fixtures
+
+* requre
+
+* VCRpy
+
+* Betamax
+
+
+## VCRpy vs. requre
+
+| A feature | VCRpy | requre |
+|-----------|-------|--------|
+| Record & replay for python requests | ✓ | ✓ |
+| Match on call stack | ✘ | ✓ |
+| Match on URL & request type | ✓ | ✘ |
+| Calling the same URL twice works as expected | ✓ | ✘ |
+| Record & replay for arbitrary functions | ✘ | ✓ |
+| Filtering sensitive data out of recorded responses | ✓ [1] | ✓ [2] |
+| Easy to use | ✓ [3] | ✘ [4] |
+| Actively maintained | ✘ | ✓ |
```

Comment:
What does this mean? I see that VCRpy is [looking for maintainers](https://github.com/kevin1024/vcrpy/issues/469), but does that meant that is not actively maintained?

Reply:
I would not say "unmaintained", but we don't know. Here is a response in the issue I linked above:

> Hi, I just wanted to update active PRs and issues to let you know I will no longer be an active maintainer on this project.
>
> I recently changed jobs where I no longer use VCRpy or Python at all so I don't have the bandwidth anymore. I won't be monitoring issues or PRs either.
>
> I will have to delegate to other maintainers or ask for some contributors to step up.
> 
> I know the Azure CLI uses VCRpy so it would be nice if they could spare some sprint time to maintain this project as it was invaluable in our CI when I was using it.
>
> Kind regards,
> Josh

Reply:
`Latest commit accffa8 on 20 Dec 2019` and the author clearly states he's no longer using it nor interested in working on it - that's a definition of a no longer actively maintained project for me

maybe betamax would be a better pick

Reply:
Okay, I missed this.

Isn't betmax requests-only?

Reply:
Do we use (internally) something different that requests? (It's the main difference between vcrpy and betamax.)

Reply:
we are using requests only afaik (copr client is using requests as well); status command is being done in async way but not with asyncio to my knowledge